Assets/scripts/Achievements/ServiceAchievements.cs

```diff
@@ -0,0 +1,67 @@
+﻿using ScriptableObjects;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+namespace Achievements
+{
+    public class ServiceAchievements : SingletonScene<ServiceAchievements>
+    {
+
+        [SerializeField] private Achievement[] achievements;
+        public event Action<string> OnAchievementUnlocked;
+        Dictionary<AchievementTypes, Achievement> gameAchievements = new Dictionary<AchievementTypes, Achievement>();
+        public override void OnInitialize()
+        {
+            base.OnInitialize();
+            GameApplication.Instance.OnPlayerSpawn += AddListener;
+
+        }
+
+
+        public void Start()
+        {
+            ///GameApplication.Instance.OnPlayerSpawn += AddListener;
+
+            foreach (Achievement achievement in achievements)
+            {
+                gameAchievements.Add(achievement.achievementType, achievement);
+            }
+           
+        }
+        public void AddListener(ControllerPlayer player)
+        {
+            PlayerData playerData = player.playerData;
+            player.OnBulletShot += AchievementUpdate;
+            player.OnUIUpdate += AchievementUpdate;
+            player.OnEnemyKilled += AchievementUpdate;
+            //Debug.Log("All listeners added for achievements");
+            if (player.GetGameStarted())
```

Comment:
i don't understand why achievement types is dependent on player.getgamesstarted ?

Reply:
get game started is to check that player has started the game not respawned